### DISCUSSION TITLE
CSS Vulnerability in PHP Login Page Affecting Username & Password Inputs

### DISCUSSION THREAD

**Discussion by GrzNdr** [community/community#123293] [Programming Help]
### Body

Hello GitHub Community,

I'm currently developing a login page using PHP and have encountered a CSS-related vulnerability with the username and password input fields. I'm reaching out for advice or solutions on how to address this issue securely.

**Issue Description:**

The vulnerability manifests as [describe the vulnerability in detail, e.g., input field manipulation through CSS, potential for CSS-based phishing, etc.]. This occurs because [explain why this happens, possibly due to external CSS files or inline styles that are user-controllable].

**Steps to Reproduce:**

1. Visit the login page at [URL or localhost setup].
2. Insert [specific malicious CSS or actions] into the username/password field.
3. [Describe what happens next, e.g., the layout changes unexpectedly, overlays appear, etc.]

**Code Snippet:**
```php
               <div class="form-item">
                    <input type="text" name="username" placeholder="Identifient" value="<?= $login ?>">
                    <input type="password" name="password" placeholder="Mot de passe" value="<?= $password ?>">
                    <p style="font-size: 12px;color:red;"><?= $erreur ?></p>
                </div>
```

**Attempted Solutions:**

- I have tried sanitizing input on both client and server sides but am unsure if my methods are robust.
- Restricted CSS sources in the page headers.

**Questions:**

1. How can I effectively prevent users from injecting malicious CSS into these input fields?
2. Are there recommended security practices for handling CSS in user input fields on a PHP login page?

Any guidance, code snippets, or resources you could provide would be greatly appreciated. Thank you for your time and help!

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by elfarfachi** [comment]
**This is the optimized version of the code snippet:**

```php
<div class="form-item">
                    <input type="text" name="username" placeholder="Identifient" value="<?= htmlspecialchars($login) ?>">
                    <input type="password" name="password" placeholder="Mot de passe" value="<?= htmlspecialchars($password) ?>">
                    <p style="font-size: 12px;color:red;"><?= htmlspecialchars($erreur) ?></p>
                </div>
```

**Reply by GrzNdr** [reply]
Thank you for the optimized code snippet! Using htmlspecialchars is a smart way to enhance security. I really appreciate your help.
